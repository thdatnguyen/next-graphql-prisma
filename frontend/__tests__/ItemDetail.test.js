import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import wait from 'waait';
import { act } from 'react-dom/test-utils';
import { MockedProvider } from '@apollo/react-testing';

import ItemDetail, {
  ITEM_DETAIL_QUERY,
} from '../components/ItemDetail.component';

import { fakeItem } from '../lib/testUtils';

describe('<ItemDetail />', () => {
  it('renders with loading status', async () => {
    const mocks = [
      {
        request: { query: ITEM_DETAIL_QUERY, variables: { id: '123' } },
        result: {
          data: {
            item: fakeItem(),
          },
        },
      },
    ];
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <ItemDetail id="123" />
      </MockedProvider>
    );
    expect(wrapper.text()).toContain('Loading...');
  });

  it('renders with proper data in H2 Tag, Image Tag and Paragraph Tag', async () => {
    const mocks = [
      {
        request: { query: ITEM_DETAIL_QUERY, variables: { id: '123' } },
        result: {
          data: {
            item: fakeItem(),
          },
        },
      },
    ];
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <ItemDetail id="123" />
      </MockedProvider>
    );
    await act(async () => wait());
    wrapper.update();
    expect(toJSON(wrapper.find('h2'))).toMatchSnapshot();
    expect(toJSON(wrapper.find('img'))).toMatchSnapshot();
    expect(toJSON(wrapper.find('p'))).toMatchSnapshot();
  });
  it('renders with Error: a not found item', async () => {
    const mocks = [
      {
        request: { query: ITEM_DETAIL_QUERY, variables: { id: '123' } },
        result: {
          errors: [{ message: 'Items not found!' }],
        },
      },
    ];
    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <ItemDetail id="123" />
      </MockedProvider>
    );
    await act(async () => wait());
    wrapper.update();
    const item = wrapper.find('[data-test="graphql-error"]');
    expect(item.text()).toContain('Items not found!');
    expect(toJSON(item)).toMatchSnapshot();
  });
});
