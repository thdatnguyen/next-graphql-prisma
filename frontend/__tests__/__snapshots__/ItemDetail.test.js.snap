// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ItemDetail /> renders with Error: a not found item 1`] = `
<p
  data-test="graphql-error"
>
  <strong>
    Shoot!
  </strong>
  Items not found!
</p>
`;

exports[`<ItemDetail /> renders with proper data in H2 Tag, Image Tag and Paragraph Tag 1`] = `
<h2>
  Viewing 
  dogs are best
</h2>
`;

exports[`<ItemDetail /> renders with proper data in H2 Tag, Image Tag and Paragraph Tag 2`] = `
<img
  alt="dogs are best"
  src="dog.jpg"
/>
`;

exports[`<ItemDetail /> renders with proper data in H2 Tag, Image Tag and Paragraph Tag 3`] = `
<p>
  dogs
</p>
`;
